### PR TITLE
Update PR template to mention user-facing changes

### PR DESCRIPTION
**Pull request by rgwood**
Draft PR for consideration: update the PR template to streamline it a bit and also mention review suggestions for user-facing changes.

@dandavison Let me know if this fits what you were thinking about

### CODE REVIEW

**Comment by dandavison** [comment]
Yes, these look like good additions and simplifications to me.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
-- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.
+- [ ] Add tests that cover your changes (either in the command examples, the crate/tests folder, or in the /tests folder)
+  - Try to think about corner cases and various ways how your changes could break. Cover those in the tests
 
 Make sure you've run and fixed any issues with these commands:
 
 - [ ] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
 - [ ] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
-- [ ] `cargo test --workspace --features=extra` to check that all the tests pass
```
Comment (sholderbach) [inline .github/pull_request_template.md:17]: I am not sure if having the boxes to manually check really adds something but a distraction. CI will tell us anyways so maybe we just keep the commands for reference.

Comment (rgwood) [reply .github/pull_request_template.md]: Good call. Removed the checkboxes.

**Comment by webbedspace** [comment]
While on the subject of updating the PR template… I've had a glance at the current command tests, and they seem a little lackluster. Instead of just 'corner cases', I think, at a minimum, it should ask that the tests cover:
- every intended shape of input (list, record, table, etc.) to the command is confirmed to 'work'
- every flag is confirmed to 'work'
- every positional is confirmed to 'work'
- every unique error message string explicitly written in the implementation (not invalid flag errors or the like) is confirmed to be output when given an intended invalid input

Not many of the existing tests seem to do this (for instance, there are zero tests that confirm `each` works with ranges (like in `2..4 | each` even though it currently does).

**Comment by rgwood** [comment]
IMO adding all 4 of those requirements would be a little too prescriptive, but I agree that we should do a better job of covering more scenarios with tests. Tracking code coverage could be another way to address the issue.

Gonna merge this for now and we can discuss test requirements separately.
